File: src/jv/item/Item.java
package jv.item;

import java.util.Random;

public class Item implements BaseItem {
    private static Random itemRandom = new Random();
    public static BaseItem[] randomItemArray(int arraySize) {
        final BaseItem[] finalArray = new BaseItem[arraySize];
        for (int i = 0; i < arraySize; i++) {
            finalArray[i] = itemRandom.nextBoolean() ?
                    new Item((short )itemRandom.nextInt(15_000), itemRandom.nextInt(250_000) + 1)
                    : INVALID_ITEM.getInstance();
        }
        return finalArray;
    }

    private final short itemId;
    private final int itemAmount;

    public Item(short itemId, int itemAmount) {
        this.itemId = itemId;
        this.itemAmount = itemAmount;
    }

    public short getItemId() { return itemId; }
    public int getItemAmount() { return this.itemAmount; }

    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (!(obj instanceof BaseItem)) return false;
        return (this.itemId == ((BaseItem) obj).getItemId()) && (this.itemAmount == ((BaseItem) obj).getItemAmount());
    }

    @Override
    public String toString() { return "Item(" + itemId + ", " + itemAmount + ")"; }
}

File: src/kt/Item.kt
package kt

import kotlin.random.Random

interface BaseItem {
    val itemId: Short
    val itemAmount: Int
    val isInvalidItem: Boolean
        get() = this is INVALID_ITEM || (itemId.toInt() == -1)
}

data class Item(override val itemId: Short, override val itemAmount: Int): BaseItem

object INVALID_ITEM: BaseItem {
    override val itemId: Short = -1
    override val itemAmount: Int = 0
    override fun toString() = "INVALID_ITEM"
}

fun randomItemArray(size: Int, maxId: Int = 15_000, maxAmount: Int = 250_000): Array<BaseItem> = Array(size) {
    if (Random.nextBoolean())
        Item(Random.nextInt(0, maxId).toShort(), Random.nextInt(1, maxAmount))
    else
        INVALID_ITEM
}